packages/@okta/vuepress-site/code/angular/okta_angular_auth_js/index.md

```diff
@@ -124,16 +124,16 @@ export class OktaAuthService {
   async handleAuthentication() {
     const tokenContainer = await this.oktaAuth.token.parseFromUrl();
 
-    this.oktaAuth.tokenManager.add('idToken', tokenContainer.tokens.idToken);
-    this.oktaAuth.tokenManager.add('accessToken', tokenContainer.tokens.accessToken);
+    this.oktaAuth.tokenManager.add('idToken', tokenContainer.tokens.idToken!);
+    this.oktaAuth.tokenManager.add('accessToken', tokenContainer.tokens.accessToken!);
 
     if (await this.isAuthenticated()) {
-      this.observer.next(true);
+      this.observer?.next(true);
     }
 
     // Retrieve the saved URL and navigate back
     const url = sessionStorage.getItem('okta-app-url');
-    this.router.navigateByUrl(url);
+    this.router.navigateByUrl(url!);
```

Comment:
what's `!` for?

Reply:
Without this I get lint error:
```
error TS2345: Argument of type 'string | null' is not assignable to parameter of type 'string | UrlTree'.
  Type 'null' is not assignable to type 'string | UrlTree'.
```

Reply:
probably because of I am not very familiar with `!` syntax, what's that means? ignore types? 

you probably can try cast it like `const url = sessionStorage.getItem('okta-app-url') as string;`, or give it a default value.

Reply:
Changed to `as` cast.
Doc about `!`: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator